### PR TITLE
Simplify if statement and syntax

### PR DESCRIPTION
**Pull request by Benny-Nottonson**
I have simplified the if statement and removed the size field in the Attribute struct. This change improves the code readability and removes unnecessary code.

### CODE REVIEW

**Comment by andresnowak** [comment]
The size value is something necessary, to be able to represent the size the same way for tensorshape, statictuple and others if they are added (so as to be able to call them the same type, in this case INTS), but everything else is okay, feel stupid that I didn't know there was a is_floating_point call haha

File: basalt/autograd/attributes.mojo [hunk 1]
```diff
@@ -83,7 +83,6 @@ struct Attribute(Stringable, CollectionElement):
     var data: Bytes[MAX_DATA_BYTES]
     var data_shape: StaticIntTuple[MAX_RANK]
     var type: AttributeType
-    var size: Int
```
Comment (andresnowak) [inline basalt/autograd/attributes.mojo:86]: This value is necessary, to be able to represent things like statictuple and tensorshape the same way (in the end they are all lists of a value, so they are the same)

**Comment by Benny-Nottonson** [comment]
okay, I actually like this as it might help with simplifying the code. I'll give this another look when i'm home today.

**Comment by Benny-Nottonson** [comment]
What about storing Size/Type/other MetaData in a seperate AttrributeMetaData struct as one thing that way we don't have wasted fields for certain data types?

**Comment by Benny-Nottonson** [comment]
for example, Scalar we need the Type information stored, but size won't matter at all, Shape we need Size but type is Int, if we add say Tensor later, we need Type and Size, but Int we need neither

**Comment by andresnowak** [comment]
Something like an attribute metadata i would say is to much I think, and the only thing needed really here is type and size, like we would never have a tensor attribute really that will always be an input. The only type of attributes one can have are ints, floats, bool, string and a list of ints, floats and strings. So tensor shape still has a size because it is a list. Like even the attribute how it works right now was designed because of the problems we have with compile time. Like in reality it could just have been a list[variant], with a name and that's all.
But yeah having the size for float, int, bool and string is something extra one could say, but its just one line really, and metadata would be more extra I think, the only metadata is type and size and that can be included in the Attribute struct

**Comment by Benny-Nottonson** [comment]
KK, ill give it a second pass right now

**Comment by Benny-Nottonson** [comment]
Ready for another check @andresnowak 👍

**Review by andresnowak — APPROVED** [review]
I think it is okay now

**Pull request merged**